pkg/autodiscovery/configresolver.go

```diff
@@ -184,11 +186,13 @@ func (cr *ConfigResolver) resolve(tpl integration.Config, svc listeners.Service)
 	}
 
 	// store resolved configs in the AC
-	cr.ac.m.Lock()
```

Comment:
Are we removing this one to avoid locking and getting stuck in the resolution ?

Reply:
I honestly don't know why this was done, @JulienBalestra? 🚌

Reply:
I'll revert this change for saafety

Reply:
Fixed with the store now 👍